### PR TITLE
Bento 1112: FE-Facet-Sorting Feature

### PR DESCRIPTION
**Pull request by iksheth**
## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

### CODE REVIEW

File: src/pages/dashboardTab/store/dashboardReducer.js [hunk 1]
```diff
@@ -400,6 +403,41 @@ function toggleCheckBoxWithAPIAction(payload, currentAllFilterVariables) {
...
+}
+/**
+ * Sort checkboxes by Count
+ *
+ * @param {object} checkboxData
+ * @return {json}
+ */
+
+function sortByCheckboxItemsByCount(checkboxData) {
+  checkboxData.sort((a, b) => a.subjects - b.subjects);
```
Comment (AjayDoddapaneni) [inline src/pages/dashboardTab/store/dashboardReducer.js:437]: Can you make it in descending order, already checked with Dr. Benson on this he wants it to be in descending order.

File: src/pages/dashboardTab/store/dashboardReducer.js [hunk 2]
```diff
@@ -654,6 +724,40 @@ const reducers = {
 
       } : { ...state };
   },
+  SORT_SINGLE_GROUP_CHECKBOX: (state, item) => {
+    const groupData = state.checkbox.data.filter((d) => item.groupName === d.groupName)[0];
```
Comment (AjayDoddapaneni) [inline src/pages/dashboardTab/store/dashboardReducer.js:728]: Can you make iterators more readable

File: src/pages/dashboardTab/store/dashboardReducer.js [hunk 3]
```diff
@@ -654,6 +724,40 @@ const reducers = {
...
+    let { sortByList } = state;
+    sortByList = sortByList || {};
+    const checkboxItems = item.sortBy === 'count'
+      ? sortByCheckboxItemsByCount(groupData.checkboxItems)
+      : sortByCheckboxItemsByAlphabet(groupData.checkboxItems);
+    const data = state.checkbox.data.map((d) => {
```
Comment (AjayDoddapaneni) [inline src/pages/dashboardTab/store/dashboardReducer.js:734]: same here with iterator

File: src/pages/dashboardTab/store/dashboardReducer.js [hunk 4]
```diff
@@ -654,6 +724,40 @@ const reducers = {
...
+    });
+
+    return { ...state, checkbox: { data }, sortByList };
+  },
+  SORT_ALL_GROUP_CHECKBOX: (state) => {
+    const { sortByList = {} } = state;
+    const { data } = state.checkbox;
+
+    data.map((d) => {
+      const dataItem = d;
```
Comment (AjayDoddapaneni) [inline src/pages/dashboardTab/store/dashboardReducer.js:751]: Same here with iterator

**Review by AjayDoddapaneni — CHANGES_REQUESTED** [review]
Very good work Karan, few minor changes

File: src/pages/dashboardTab/store/dashboardReducer.js [hunk 5]
```diff
@@ -725,20 +725,23 @@ const reducers = {
       } : { ...state };
   },
   SORT_SINGLE_GROUP_CHECKBOX: (state, item) => {
-    const groupData = state.checkbox.data.filter((d) => item.groupName === d.groupName)[0];
+    const groups = state.checkbox.data;
```
Comment (AjayDoddapaneni) [inline src/pages/dashboardTab/store/dashboardReducer.js:728]: One small change, we don't need a separate variable    "groups" as we are not reusing it. I know it's for readability but the "group" iterator in the filter can provide enough.

Comment (iksheth) [reply src/pages/dashboardTab/store/dashboardReducer.js:728]: Sure. Updating.

Comment (iksheth) [reply src/pages/dashboardTab/store/dashboardReducer.js]: Updated !

**Comment by AjayDoddapaneni** [comment]
Looks good

**Review by AjayDoddapaneni — APPROVED** [review]

**Review by skiran86 — APPROVED** [review]

**Pull request merged**